### DISCUSSION TITLE
How to insert only when rows are not present

### DISCUSSION THREAD

**Discussion by amanv8060** [simolus3/drift#989] [Q&A] [answered]
Continued from #988
@simolus3  i tried that as well , it ends up adding  another question  

and i think its just a repetition 

```
   void toggleFavorite(Question question) {
    Question newQues = Question(
        isFavourite: !question.isFavourite,
        title: question.title,
        code: question.code,
        question: question.question,
        uid: question.uid,
        category: question.category);
    print(newQues.toString());
    update(questions).replace(newQues);
  }
```
toggles favorite for both of them

**Comment by simolus3** [comment, marked as answer]
In sqlite, primary keys are unique. So I'm not really sure how the update could affect more than one row here. Does it work if you remove the `customConstraint()` from `uid`? It shouldn't be necessary since `uid` is a primary key already.

**Reply by amanv8060** [reply]
`Flutter clean` solved it  , idk why
